# Investigate Titanic's Data
## Questions to ask ourselves
### What factors made people more likely to survive?
* Sex
* Class
* Age
* How much they paid

In [1]:
#imports
import pandas as pd
import numpy as np

In [22]:
raw_data = pd.read_csv('titanic_data.csv')

In [26]:
raw_data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex  Age  SibSp  \
0                            Braund, Mr. Owen Harris    male   22      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female   38      1   
2                             Heikkinen, Miss. Laina  female   26      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   35      1   
4                           Allen, Mr. William Henry    male   35      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

## Data Wrangling
We need to find the amount of nulls that our data has.

`describe` function might be useful

In [9]:
raw_data.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

we realise however that in this way we are not able to see NA in non-numeric columns.

We move to another option:

In [12]:
raw_data.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

### How do we treat nulls
#### In AGE
Out of 891 rows, we have 177 *NaN*, which represent roughly a 20%. If we replace this NaN with some other value we should be guard value, so it does not affect the rest of the values. 

### In Cabin
Out of 891 rows, 687 are nulls, representing an astounding 77%. **Ignoring this column** altogether makes more sense.

### In Embarked
Only 2 NaN in this column make it possible to simply **ignore this rows**. We could also decide another value and see how they behave.

### Code
#### Age

In [24]:
clean_data = raw_data.copy()
clean_data['Age'] = clean_data['Age'].fillna(-1)

#### Cabin

In [25]:
clean_data.drop('Cabin', axis=1, inplace=True)

#### Embarked
Before deleting anything, let's check the rows

In [27]:
raw_data[raw_data['Embarked'].isnull()]

PassengerId  Survived  Pclass                                       Name  \
61            62         1       1                        Icard, Miss. Amelie   
829          830         1       1  Stone, Mrs. George Nelson (Martha Evelyn)   

        Sex  Age  SibSp  Parch  Ticket  Fare Cabin Embarked  
61   female   38      0      0  113572    80   B28      NaN  
829  female   62      0      0  113572    80   B28      NaN

It looks a bit strange that they both survived, are in the same Cabin and we lack their Embarked information, using the same ticket.

Instead of deleting them we will leave the rows for now.

## Data Exploration

In [ ]:
Survival by age